packages/react/src/Switch/Switch.test.tsx

```diff
@@ -0,0 +1,47 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+import { createRef } from 'react'
+import { Switch } from './Switch'
+import '@testing-library/jest-dom'
+
+describe('Switch', () => {
+  it('renders an element with role switch', () => {
+    render(<Switch />)
+    const switchElement = screen.getByRole('switch')
+
+    expect(switchElement).toBeInTheDocument()
+  })
+
+  it('renders the Switch with the default state unchecked', () => {
+    render(<Switch />)
+    const switchElement = screen.getByRole('switch')
+
+    expect(switchElement).not.toBeChecked()
+  })
+
+  it('sets the checked state when switch is clicked', () => {
+    render(<Switch />)
+    const switchElement = screen.getByRole('switch')
+
+    expect(switchElement).not.toBeChecked()
+
+    fireEvent.click(switchElement)
+
+    expect(switchElement).toBeChecked()
+  })
+
+  it('renders a design system BEM class name', () => {
+    const { container } = render(<Switch />)
+    const component = container.querySelector(':only-child')
+
+    expect(component).toHaveClass('amsterdam-switch')
+  })
+
```

Comment:
We can add a couple more test cases here:

- it can have a additional class name
- it is not disabled by default
- it can have a disabled state
- it can trigger a change event
- it does not trigger a change event when disabled
- it triggers a change event by clicking on label

You can mostly copy-paste these from the Checkbox component I think